src/variety/railpool.js

```diff
@@ -0,0 +1,659 @@
+					}
+					off = false;
+				}
+
+				if (!bd.cell[c]) {
+					break;
+				}
+			}
+			this.outbstr = bstr.substr(i + 1);
+		},
+		encodeNumber_railpool: function() {
+			var count = 0,
+				cm = "",
+				bd = this.board;
+			for (var c = 0; c < bd.cell.length; c++) {
+				var pstr = "",
+					qn = bd.cell[c].qnums;
+
+				if (qn.length > 0) {
+					for (var n = 0; n < qn.length; n++) {
+						var curNum = qn[n],
+							chars = [];
+						while (curNum > 0) {
+							var curDigit = curNum % 10;
+							curNum = Math.floor(curNum / 10);
+							chars.push((curNum < 10 ? curDigit : curDigit + 10).toString(36));
+						}
+
+						if (chars.length === 0) {
+							pstr += "0";
+						} else {
+							pstr += chars.reverse(chars).join("");
+						}
+					}
+				} else {
+					count++;
+				}
+
+				if (count === 0) {
+					cm += pstr;
+					count = 1;
+				} else if (pstr !== "" || count === 16) {
+					cm += (19 + count).toString(36) + pstr;
+					count = pstr === "" ? 0 : 1;
+				}
+			}
+			if (count > 1) {
+				cm += (19 + count).toString(36);
+			}
+
+			this.outbstr += cm;
+		}
+	},
+	//---------------------------------------------------------
+	FileIO: {
+		decodeData: function() {
+			this.decodeQnums_railpool();
+			this.decodeEmpty();
+			this.decodeAreaRoom();
+			this.decodeBorderLine();
+		},
+		encodeData: function() {
+			this.encodeQnums_railpool();
+			this.encodeEmpty();
+			this.encodeAreaRoom();
+			this.encodeBorderLine();
+		},
+		// from tapaloop.js
+		decodeQnums_railpool: function() {
+			this.decodeCell(function(cell, ca) {
+				if (ca !== ".") {
+					cell.qnums = [];
+					var array = ca.split(/,/);
+					for (var i = 0; i < array.length; i++) {
+						cell.qnums.push(array[i] !== "-" ? +array[i] : -2);
+					}
+				}
+			});
+		},
+		encodeQnums_railpool: function() {
+			this.encodeCell(function(cell) {
+				if (cell.qnums.length > 0) {
+					var array = [];
+					for (var i = 0; i < cell.qnums.length; i++) {
+						array.push(cell.qnums[i] >= 0 ? "" + cell.qnums[i] : "-");
+					}
+					return array.join(",") + " ";
+				} else {
+					return ". ";
+				}
+			});
+		},
+		// from country.js
+		decodeEmpty: function() {
+			this.decodeCell(function(cell, ca) {
+				if (ca === "*") {
+					cell.ques = 7;
+				}
+			});
+		},
+		encodeEmpty: function() {
+			this.encodeCell(function(cell) {
+				if (cell.ques === 7) {
+					return "* ";
+				} else {
+					return ". ";
+				}
+			});
+		}
+	},
+
+	//---------------------------------------------------------
+	// 正解判定処理実行部
+	AnsCheck: {
+		checklist: [
+			"checkBranchLine",
+			"checkCrossLine",
+
+			"checkUncluedLength",
+			"checkMissingLength",
+			"checkIncompatibleSets",
+
+			"checkDeadendLine+",
+			"checkOneLoop",
+			"checkNoLine"
+		],
+
+		checkUncluedLength: function() {
+			var result = true,
+				rooms = this.board.roommgr.components;
+			for (var r = 0; r < rooms.length; r++) {
+				var clist = rooms[r].clist,
+					clueSet = rooms[r].clist.getClueSet();
```

Comment:
When I got this error, I initially wasn't sure which region's numbers were causing it. Could you highlight the entire region for this check, like in `checkIncompatibleSets`?

Reply:
Done, but note that Lohkous doesn't have this feature.

Reply:
Lohkous doesn't need the feature, since the error will only show up if the region is completed, and the highlight will be contained entirely within that region. But for Rail Pool, errored line segments can cover any amount of regions.

Reply:
Ah, I see that the region is now highlighted when a number is unused, but not when a segment is invalid. I'd like it to be the other way around.

Reply:
Oops, that makes more sense - fixed now.